server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -790,8 +791,10 @@ public String startVerification() {
                 }
                 return seed;
             }
-        } catch (IOException exp) {
-            throw new RepositoryVerificationException(metadata.name(), "path " + basePath() + " is not accessible on master node", exp);
+        } catch (Exception e) {
+            throw new RepositoryVerificationException(metadata.name(),
+                        String.format(Locale.ENGLISH, "Unable to verify the repository, [%s] is not accessible on master node: %s '%s'",
```

Comment:
This seems strange on the first view. The problem with propagating the underlying exception and handling the message creation later is, that the root cause might end up in a `NotSerializableExceptionWrapper` if it is not serializable across nodes and then error message will differ whenever it was wrapped or not. Now the error message is created before the exception is further propagated.